### ISSUE TITLE
Usage over time

### ISSUE THREAD

**Issue by erikhuck** [MoseleyBioinformaticsLab/gpu_tracker#1]
While the tracker currently reports the maximum usage of each resource, a user may want to see the usage over unit time (whichever sleep time they choose or the default). This should not be the default behavior considering it would use up more space (i.e. an increasing amount of space the longer the context lasts). So it should be an option. We may want to append to a file considering (e.g. a csv file) considering the increase in space per unit time. That should also be an option since a user may want the results provided in memory.

**Comment by erikhuck** [comment]
Some things to keep in mind about the above comment is that the Tracker is already designed to handle specific code blocks and is already computing the mean and max of those code blocks. Instead of calling subprofile, the user could just have multiple calls to the tracking. The use case here is more if they wanted to profile both a larger code block and a smaller code block within it. They actually can accomplish this currently but they'd have two tracker subprocesses running simultaneously which may not be necessary and uses up extra resources. But if they really want to, they can. Alternatively, we could have the subprofile method be part of the tracker object itself which, when called, sends a signal to the already extant subprocess to inform it to start tracking an additional resource usage object for the subsection of the code block. This would bypass the need to keep two files, including a database, if the user is just wanting to get mean/max values anyway which we already have an implementation for and we are probably better off re-using that than creating a rather complicated solution.

The challenge though is the CLI since this granular control given this alternative implementation would only be accessible via the API. What we could do is have an entirely different option that is just for creating the SQL database if that's what the user wants. They could also have the choice to store it in a CSV file instead. Then the user can do what they wish with that SQL database. Since they would have to construct their own timestamps anyway, they can do so in whatever way they please and the database (or CSV) file they decide to create would just include the timestamps and the measurements and they can use that data in whatever way suites them.

In summary, the subprofile method in the proposed implementation above is only available in the API and when using the API, the user already largely has the capability to do what it offers using the current implementation. They can even do profiling within profiling with the current implementation though if we wanted to avoid creating a whole other tracking subprocess, we could make a slight adjustment to the current implementation rather than a major overhaul. The benefits of the proposal above is more for the CLI but the CLI has limitations such that the user would have to do some work anyway in which case we could have an entirely separate feature for creating the database or CSV and they can use that file as they please.
